### DISCUSSION TITLE
Azure appservice deployment

### DISCUSSION THREAD

**Discussion by nidalasw** [adonisjs/core#4520] [Help]
Hello everyone,

I'm new to AdonisJS 6 and I'm currently attempting to deploy a basic API on Azure App Service. The deployment via the Azure plugins in VSCode was successful, but when I try to access my API, I encounter the following error:
:( Application Error
Attached are the logs. Is there something I'm missing?

Thank you for your assistance!

**Logs**
INFO  - Starting container for site
INFO  - docker run -d --expose=80 --name api-digimall-gardencity-adonis_0_991f1d24 -e WEBSITE_USE_DIAGNOSTIC_SERVER=true -e WEBSITES_PORT=3333 -e PORT=80 -e WEBSITE_SITE_NAME=api-digimall-gardencity-adonis -e WEBSITE_AUTH_ENABLED=False -e WEBSITE_ROLE_INSTANCE_ID=0 -e WEBSITE_HOSTNAME=api-digimall-gardencity-adonis.azurewebsites.net -e WEBSITE_INSTANCE_ID=d8ce030d10a8488f5af05adb44862a92568cc382a56095090ac9f1a788fb1dff -e HTTP_LOGGING_ENABLED=1 -e NODE_OPTIONS=--require /agents/nodejs/build/src/Loader.js appsvc/node:20-lts_20240207.3.tuxprod node build/bin/server.js
INFO  - Initiating warmup request to container api-digimall-gardencity-adonis_0_991f1d24 for site api-digimall-gardencity-adonis
INFO  - Waiting for response to warmup request for container api-digimall-gardencity-adonis_0_991f1d24. Elapsed time = 16.9781533 sec
INFO  - Waiting for response to warmup request for container api-digimall-gardencity-adonis_0_991f1d24. Elapsed time = 33.6936937 sec
INFO  - Waiting for response to warmup request for container api-digimall-gardencity-adonis_0_991f1d24. Elapsed time = 49.101963 sec
INFO  - Waiting for response to warmup request for container api-digimall-gardencity-adonis_0_991f1d24. Elapsed time = 64.6769662 sec
INFO  - Waiting for response to warmup request for container api-digimall-gardencity-adonis_0_991f1d24. Elapsed time = 82.811975 sec
INFO  - Waiting for response to warmup request for container api-digimall-gardencity-adonis_0_991f1d24. Elapsed time = 98.4616748 sec
INFO  - Waiting for response to warmup request for container api-digimall-gardencity-adonis_0_991f1d24. Elapsed time = 113.8471751 sec
INFO  - Waiting for response to warmup request for container api-digimall-gardencity-adonis_0_991f1d24. Elapsed time = 129.8260236 sec
INFO  - Waiting for response to warmup request for container api-digimall-gardencity-adonis_0_991f1d24. Elapsed time = 145.1895016 sec
INFO  - Waiting for response to warmup request for container api-digimall-gardencity-adonis_0_991f1d24. Elapsed time = 160.4492765 sec
INFO  - Waiting for response to warmup request for container api-digimall-gardencity-adonis_0_991f1d24. Elapsed time = 175.7839159 sec
INFO  - Waiting for response to warmup request for container api-digimall-gardencity-adonis_0_991f1d24. Elapsed time = 191.1248862 sec
INFO  - Waiting for response to warmup request for container api-digimall-gardencity-adonis_0_991f1d24. Elapsed time = 206.5878133 sec
INFO  - Waiting for response to warmup request for container api-digimall-gardencity-adonis_0_991f1d24. Elapsed time = 221.8861038 sec
ERROR - Container api-digimall-gardencity-adonis_0_991f1d24 for site api-digimall-gardencity-adonis did not start within expected time limit. Elapsed time = 230.1408088 sec
ERROR - Container api-digimall-gardencity-adonis_0_991f1d24 didn't respond to HTTP pings on port: 80, failing site start. See container logs for debugging.
INFO  - Stopping site api-digimall-gardencity-adonis because it failed during startup.
`

**Comment by RomainLanz** [comment]
Hey @nidalasw! 👋🏻

It seems that your container is not accessible, are you sure it is listening on port `80` and host `0.0.0.0`?

**Reply by nidalasw** [reply]
Hi Romain,
Thanks for your message
I didnt use a container it is the default container used by Azure AppService. What i did is I created an app service via the UI of Azure.
It looks like it is a port issue as you said. I tried to add an ENV varibale PORT:3333 with no luck

![image](https://github.com/adonisjs/core/assets/78143153/65989509-ff87-4cf6-8ec8-9c9e07466f5b)

Thanks for your help!

**Reply by RomainLanz** [reply]
Have you added an env variable `HOST` with the value `0.0.0.0`?

**Reply by nidalasw** [reply]
Yes I did
